src/txn/txn_rollback_to_stable.c

```diff
@@ -124,7 +133,73 @@ __txn_abort_newer_insert(
     WT_INSERT *ins;
 
     WT_SKIP_FOREACH (ins, head)
-        __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
+        if (ins->upd != NULL)
+            __txn_abort_newer_update(session, ins->upd, rollback_timestamp);
+}
+
+/*
+ * __txn_abort_row_ondisk_kv --
+ *     Fix the on-disk row K/V version according to the given timestamp.
+ */
+static int
+__txn_abort_row_ondisk_kv(
+  WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip, wt_timestamp_t rollback_timestamp)
+{
+    WT_CELL_UNPACK *vpack, _vpack;
+    WT_DECL_RET;
+    WT_ITEM buf;
+    WT_PAGE_MODIFY *mod;
+    WT_UPDATE *upd, **upd_entry;
+    size_t size;
+
+    vpack = &_vpack;
+    __wt_row_leaf_value_cell(session, page, rip, NULL, vpack);
+    if (vpack->start_ts > rollback_timestamp) {
+        /* No visible updates in the history store, may be a newly inserted key. Remove it. */
+        WT_RET(__wt_update_alloc(session, NULL, &upd, &size, WT_UPDATE_TOMBSTONE));
+        upd->txnid = WT_TXN_NONE;
+        upd->durable_ts = WT_TS_NONE;
+        upd->start_ts = WT_TS_NONE;
+    } else if (vpack->stop_ts != WT_TS_MAX && vpack->stop_ts > rollback_timestamp) {
+        /*
+         * Clear the remove operation from the key by inserting the original on-disk value as a
+         * standard update.
+         */
+        WT_CLEAR(buf);
+
+        /*
+         * If a value is simple and is globally visible at the time of reading a page into cache, we
```

Comment:
I do not understand the definition of "simple" in this comment. Does it mean not an overflow record? Something else?

Reply:
Simple means the values that are not compressed, I updated the comment to explain it. Those values when they are globally visible, they will be placed into the WT_ROW itself to directly reference it.